bpmn-model/src/main/java/io/camunda/zeebe/model/bpmn/validation/zeebe/ZeebeElementValidator.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.zeebe.model.bpmn.validation.zeebe;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+import org.camunda.bpm.model.xml.instance.ModelElementInstance;
+import org.camunda.bpm.model.xml.validation.ModelElementValidator;
+import org.camunda.bpm.model.xml.validation.ValidationResultCollector;
+
+public final class ZeebeElementValidator<T extends ModelElementInstance>
+    implements ModelElementValidator<T> {
+
+  private final Class<T> elementType;
+  private final List<AttributeAssertion<T>> attributeAssertions = new ArrayList<>();
+
+  private ZeebeElementValidator(final Class<T> elementType) {
+    this.elementType = elementType;
+  }
+
+  @Override
+  public Class<T> getElementType() {
+    return elementType;
+  }
+
+  @Override
+  public void validate(final T element, final ValidationResultCollector validationResultCollector) {
+
+    attributeAssertions.forEach(
+        assertions -> {
+          final String attributeValue = assertions.attributeSupplier.apply(element);
+          if (attributeValue == null || attributeValue.isEmpty()) {
```

Comment:
🔧  We should consider replacing the `isEmpty` with `isBlank`. Generally a blank string would also not be correct entry.

Reply:
Good idea. But in the model API, we are limited to Java 8. `isBlank()` was introduced with Java 11 :sweat_smile: